File: uva_11579.cpp
# include <iostream>
# include <iomanip>
# include <vector>
# include <cmath>
# include <algorithm>
using namespace std;
double area(double a, double b, double c){
    double s = (a+b+c)/2;
    return s*(s-a)*(s-b)*(s-c);
}
int main(void){
    cout.precision(2);
    int num;
    cin >> num;
    while(num--){
        int N;
        cin >> N;
        vector<double> v;
        v.resize(N);
        for (int i=0; i<N;++i)
            cin >> v[i];
        sort(v.begin(), v.end());
        double ans=0;
        for (int i=0; i+2<v.size(); ++i){
            if (v[i]+v[i+1]<=v[i+2])
                continue;
            double temp=area(v[i], v[i+1], v[i+2]);
            ans=temp>ans?temp:ans;
        }
        cout << fixed << sqrt(ans) << endl;
    }
    return 0;
}

File: uva_11579.py
import math
def area(a, b, c):
    s=(a+b+c)/2
    return s*(s-a)*(s-b)*(s-c)
num=int(input())
for _ in range(num):
    data=input().split()
    N=int(data[0])
    for i in range(N):
        data[i]=float(data[i+1])
    data=data[:-1]
    data.sort()
    ans=0.
    for i in range(N-2):
        if data[i]+data[i+1]<=data[i+2]:
            continue
        temp=area(data[i], data[i+1], data[i+2])
        ans=temp if temp>ans else ans
    ans='%.2f'%math.sqrt(ans)
    print(ans)

